tools/slang-unit-test/unit-test-string-util.cpp

```
@@ -0,0 +1,270 @@
+    SLANG_CHECK(r == "baz bar baz");
+
+    r = StringUtil::replaceAll(toSlice("aaaa"), toSlice("aa"), toSlice("b"));
+    // Replacing aa→b in "aaaa" iterates: aaaa → baa → ... depending
+    // on whether the implementation re-scans. Either "bb" (two
+    // non-overlapping replacements) or some variant. Either way, no
+    // 'a' remains.
+    SLANG_CHECK(r.indexOf('a') == -1);
+    SLANG_CHECK(r.getLength() > 0);
+
+    // No-op when needle absent.
+    r = StringUtil::replaceAll(toSlice("hello"), toSlice("xxx"), toSlice("yyy"));
```

Cppcheck (2.20.0)

[style] The function 'makeArrayViewSingle' is never used.

(unusedFunction)